Sources/GRPCHTTP2TransportNIOPosix/GRPCHTTP2TransportNIOPosix.swift

```diff
@@ -15,7 +15,188 @@
  */
 
 import GRPCCore
+@_spi(Package) import GRPCHTTP2Core
+import NIOCore
+import NIOExtras
+import NIOPosix
 
-@available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
-public struct GRPCHTTP2TransportNIOPosix {
+extension HTTP2ServerTransport {
+  @available(macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0, *)
+  public struct Posix: ServerTransport {
+    private let address: GRPCHTTP2Core.SocketAddress
+    private let config: Config
+    private let eventLoopGroup: MultiThreadedEventLoopGroup
+    private let serverQuiescingHelper: ServerQuiescingHelper
+
+    public init(
+      address: GRPCHTTP2Core.SocketAddress,
+      config: Config,
+      eventLoopGroup: MultiThreadedEventLoopGroup = .singletonMultiThreadedEventLoopGroup
+    ) {
+      self.address = address
+      self.config = config
+      self.eventLoopGroup = eventLoopGroup
+      self.serverQuiescingHelper = ServerQuiescingHelper(group: self.eventLoopGroup)
+    }
+
+    public func listen(
+      _ streamHandler: @escaping (RPCStream<Inbound, Outbound>) async -> Void
+    ) async throws {
+      let serverChannel = try await ServerBootstrap(group: self.eventLoopGroup)
+        .serverChannelInitializer { channel in
+          let quiescingHandler = self.serverQuiescingHelper.makeServerChannelHandler(
+            channel: channel
+          )
+          return channel.pipeline.addHandler(quiescingHandler)
+        }
+        .bind(to: self.address) { channel in
+          channel.eventLoop.makeCompletedFuture {
+            return try channel.pipeline.syncOperations.configureGRPCServerPipeline(
+              channel: channel,
+              compressionConfig: self.config.compression,
+              keepaliveConfig: self.config.keepalive,
+              connectionConfig: self.config.connection,
+              http2Config: self.config.http2,
+              rpcConfig: self.config.rpc,
+              useTLS: false
+            )
+          }
+        }
+
+      try await serverChannel.executeThenClose { inbound in
+        try await withThrowingDiscardingTaskGroup { serverTaskGroup in
+          for try await (connectionChannel, streamMultiplexer) in inbound {
+            serverTaskGroup.addTask {
+              try await connectionChannel.executeThenClose { connectionInbound, connectionOutbound in
+                try await withThrowingDiscardingTaskGroup { connectionTaskGroup in
+                  connectionTaskGroup.addTask {
+                    for try await _ in connectionInbound {}
+                  }
+
+                  connectionTaskGroup.addTask {
+                    try await withThrowingDiscardingTaskGroup { streamTaskGroup in
+                      for try await (http2Stream, methodDescriptor) in streamMultiplexer.inbound {
+                        streamTaskGroup.addTask {
+                          try await http2Stream.executeThenClose { inbound, outbound in
+                            let descriptor = try await methodDescriptor.get()
```

Comment:
What happens if a child task in a throwing-discarding-task-group throws? We should check this because if the stream throws it _shouldn't_ result in the connection being closed.

Same goes for the connection task group: if a connection throws we shouldn't cancel the listening channel either.